src/chart/generateCategoricalChart.tsx

```diff
@@ -2087,6 +2109,19 @@ export const generateCategoricalChart = ({
               : 'payload.'.concat(tooltipAxis.dataKey.toString());
           activePoint = findEntryInArray(points, specifiedKey, activeLabel);
           basePoint = isRange && baseLine && findEntryInArray(baseLine, specifiedKey, activeLabel);
+        } else if (tooltipAxis.dataKey && tooltipAxis.allowSelectNearestValue && tooltipAxis?.categoricalDomain) {
```

Comment:
this seems like the same logic as above? Can we modularize a bit better and extract some logic into a helper function?

Reply:
If is the same logic as above, since finding the nearest active point for the tooltip, and finding the nearest active point to visually highlight on the plot are at different parts of the code. 

In the future it probably would make sense to have these in similar places.

Happy to make 2 helper functions for each of these tasks.

Reply:
right, they can still be called in the same place as they are but just extract the duplicate logic into an (easier to test) helper function)

Reply:
Making helper function with intention revealing names would also help us not increase the complexity of this file further.